app/js/questions_controller.js

```diff
@@ -43,6 +45,29 @@
     var tab_panels = container.querySelectorAll('[role="tabpanel"]');
     toggle_state(tab_panels, 'aria-hidden');
 
+    load_active_questions();
+  }
+
+  function load_active_questions() {
+    User.get_user().then(function(response){
+      var promise = SumoDB.get_active_questions(response.username);
+      console.log(response.username);
+
+      promise.then(function(response) {
+        console.log(response);
+        var results = response.results;
+        if (results.length) {
+          display_questions(response, active_questions_list_container, load_more_active_questions_button);
+        } else {
+          var html = nunjucks.render(QUESTION_LIST_TMPL, {
+            message: MSG_NO_QUESTIONS
+          });
+          active_questions_list_container.insertAdjacentHTML('beforeend', html);
+
+          Utils.toggle_spinner();
+        }
```

Comment:
I think we can factor `load_initial_questions` and `load_active_questions` even more.
`load_initial_questions` does a bit more but it is all under conditions so we can use the same code after the HTTP request was done.
If we move the `if (results.length)` test inside a new display_initial_questions we can do something like:

```js
function load_active_questions() {
  User.get_user().then(function(user){
    return SumoDB.get_active_questions(user.username);
  }).then(function(response) {
    return display_initial_questions(response, active_questions_list_container, load_more_button);
  });
}
function load_unanswered_questions() {
  var params = Utils.get_url_parameters(location);
  if ('helper' === params.role) {
    promise = SumoDB.get_unanswered_questions();
  } else {
    promise = User.get_user().then(SumoDB.get_my_questions);
  }
  promise.then(function(response) {
    if (results.length) {
        if (!my_questions_list_container.dataset.all) {
          results = results.slice(0, 3);
        }
    }
    return display_initial_questions(response, active_questions_list_container, load_more_button);
  }).then(function() {
    var all_questions = document.getElementById('all_questions_button');
    if (all_questions) {
      all_questions.classList.remove('hide');
    }
}

function display_initial_questions(response, container, load_more_button) {
  if (response.results.length) {
    return display_questions(response, container, load_more_button);
  }
  
  var html = nunjucks.render(QUESTION_LIST_TMPL, {
    message: MSG_NO_QUESTIONS
  });
  container.insertAdjacentHTML('beforeend', html);
  Utils.toggle_spinner();
}
```